cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -3542,6 +3596,68 @@ bool CacheAllocator<CacheTrait>::startNewReaper(
   return true;
 }
 
+template <typename CacheTrait>
+auto CacheAllocator<CacheTrait>::getAssignedMemoryToBgWorker(
+    size_t workerId, size_t numWorkers) {
+  std::vector<MemoryDescriptorType> asssignedMemory;
+  auto pools = filterCompactCachePools(allocator_->getPoolIds());
+  for (const auto pid : pools) {
```

Comment:
Can we have a function return vector<vector<MemoryType>>. Then loop through this vector to assign to each worker thread, taking the lock one by one?

Reply:
Sure, done.